### PR TITLE
Update WASM image

### LINKED ISSUE #34280: LC_ALL warning during CI / PR builds
Our standard CI / PR builds all issue the following warning when building the WASM job

```
        warning Code bash(0,0): warning : (NETCORE_ENGINEERING_TELEMETRY=Build) setlocale: LC_ALL: cannot change loca
le (en_US.UTF-8)
        warning Code bash(0,0): warning : (NETCORE_ENGINEERING_TELEMETRY=Build) setlocale: LC_ALL: cannot change loca
le (en_US.UTF-8)
        warning Code bash(0,0): warning : (NETCORE_ENGINEERING_TELEMETRY=Build) setlocale: LC_ALL: cannot change loca
le (en_US.UTF-8)
        warning Code bash(0,0): warning : (NETCORE_ENGINEERING_TELEMETRY=Build) setlocale: LC_ALL: cannot change loca
le (en_US.UTF-8)
        warning Code bash(0,0): warning : (NETCORE_ENGINEERING_TELEMETRY=Build) setlocale: LC_ALL: cannot change loca
le (en_US.UTF-8)
```

These aren't fatal warnings but they do add to build output churn that makes it difficult to track down real failures. Need to remove this from the log.

### PR DESCRIPTION
**Pull request by jaredpar**
Move to an Ubuntu image which has the `en_US.UTF-8` locale available.
This should remove the spurious warning that we're seeing during
PR / CI

closes #34280

### CODE REVIEW

**Comment by jaredpar** [comment]
Once merged builds should be completely warning free.

**Comment by jaredpar** [comment]
@ViktorHofer @safern @akoeplinger

**Comment by jaredpar** [comment]
The libraries failure was a Helix 503, re-running.

**Comment by safern** [comment]
Thanks @jaredpar. Since we're now warnings free, in order to prevent introducing new warnings, could you also remove: `_warnAsErrorArg` and all it's references from: https://github.com/dotnet/runtime/blob/5369c7119b0f174c2614fe4ee197bdb0c42010a1/eng/pipelines/libraries/base-job.yml as part of this change?

**Comment by jaredpar** [comment]
> Since we're now warnings free, in order to prevent introducing new warnings, could you also remove: _warnAsErrorArg and all it's references from:

I will definitely do that but prefer to do as a separate PR. Given the last time I tried that I broke the outer loop wanted to have a PR where I just changed that, queued up a coulpe of the outer loop jobs and ensure I didn't break them with the warn as error change (again 😦 ).

**Comment by safern** [comment]
I think for this change you should be safe as we build the same configurations and same containers for outerloop jobs, but since you're making the change, I'll leave it up to you.

**Review by safern — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]

**Pull request merged**
